src/viz/RobotModel.js

```diff
@@ -100,16 +104,40 @@ class RobotModel extends URDFLoader {
   }
 
   destroy() {
+    const robotModelParent = this.robotModel && this.robotModel.parent;
+    if (robotModelParent) {
+      // TODO: proper scene clear for all viz types
+      // just parent.remove is not enough
+      Object.values(this.robotModel.links).map(linkObj => {
+        if (linkObj.parent) {
+          linkObj.parent.remove(linkObj);
+        }
+      });
+      robotModelParent.remove(this.robotModel);
+    }
+    this.robotModel = null;
     if (this.object.parent) {
       this.object.parent.remove(this.object);
     }
   }
 
   hide() {
+    if (!this.robotModel) {
+      return;
+    }
+    Object.values(this.robotModel.links).map(linkObj => {
+      linkObj.visible = false;
```

Comment:
This can have unintended consequences. Changing a visibility of link object also affects it's children.
In the use case of urdf, hiding a link might only mean hiding the meshes for that link rather than hiding the entire child tree of the link

Reply:
If the links are instances of link class from urdf-js, it can have a hide function that robot hide can call.

Reply:
Since the loop runs on all links, no child link would be left untouched.

Reply:
The child links should be left untouched right?
If a gripper is added as child of robot's end effector, hiding the link should not hide the gripper right. The intended action would be to only hide the robot model according to me.

Reply:
This problem and the other one of deleting valid world frames when deleting the robot should perhaps be fixed by namespacing the world frames like `panda-container` then `panda` is attached to it rather than become a world frame itself.

Then `urdf-js` can expose `destroy`, `show`, `hide` on individual links that destroy the mesh and remove it from the container object.